test-utils_test.go

```diff
@@ -197,6 +197,8 @@ func newTestRequest(method, urlStr string, contentLength int64, body io.ReadSeek
 		body.Seek(0, 0)
 		// Add body
 		req.Body = ioutil.NopCloser(body)
+	} else {
+		req.Body = ioutil.NopCloser(bytes.NewReader([]byte("")))
```

Comment:
@harshavardhana : There was a crash while reading the req.Body in `ioutil.ReadAll` insie `authHandler`. This fixed it.